checkpoint/recovery.go

```diff
@@ -156,30 +85,33 @@ func copyToLocalFile(ctx context.Context, logger log.Log, fs afero.Fs, dataDir,
 	return dst, nil
 }
 
+type PreservedData struct {
+	Deps   []*types.VerifiedActivationTx
+	Proofs []*types.PoetProofMessage
+}
+
 func Recover(
 	ctx context.Context,
 	logger log.Log,
 	fs afero.Fs,
 	cfg *RecoverConfig,
-	nodeID types.NodeID,
-	uri string,
-	restore types.LayerID,
-) error {
+) (*PreservedData, error) {
 	db, err := sql.Open("file:" + filepath.Join(cfg.DataDir, cfg.DbFile))
 	if err != nil {
-		return fmt.Errorf("open old database: %w", err)
+		return nil, fmt.Errorf("open old database: %w", err)
 	}
 	defer db.Close()
-	newdb, err := RecoverWithDb(ctx, logger, db, fs, cfg, nodeID, uri, restore)
+	preserve, err := RecoverWithDb(ctx, logger, db, fs, cfg)
 	if err != nil {
-		return err
-	}
-	if newdb != nil {
-		if err = newdb.Close(); err != nil {
-			return fmt.Errorf("close new db: %w", err)
+		if errors.Is(err, ErrCheckpointNotFound) {
+			logger.With().Info("no checkpoint file available. not recovering",
+				log.String("uri", cfg.Uri),
+			)
+			return nil, nil
 		}
+		return nil, err
 	}
-	return nil
+	return preserve, nil
```

Comment:
switching makes the error handling a bit easier to understand:

```golang
	preserve, err := RecoverWithDb(ctx, logger, db, fs, cfg)
	switch {
	case errors.Is(err, ErrCheckpointNotFound):
		logger.With().Info("no checkpoint file available. not recovering",
			log.String("uri", cfg.Uri),
		)
		return nil, nil
	case err != nil:
		return nil, err
	default:
		return preserve, nil
	}
```

Reply:
for me if/else block is easier to read.....but ok.